crates/ty_python_semantic/src/module_resolver/resolver.rs

```diff
@@ -330,8 +332,42 @@ impl SearchPaths {
         &self.typeshed_versions
     }
 
-    pub fn python_version(&self) -> Option<&PythonVersionWithSource> {
-        self.python_version.as_ref()
+    pub fn python_version(&self) -> Option<Cow<PythonVersionWithSource>> {
```

Comment:
Nit: Maybe `to_python_version` to signify that this operation is no longer "free"